werkzeug/_internal.py

```diff
@@ -44,8 +44,10 @@
 _quote_re = re.compile(b'[\\\\].')
 _legal_cookie_chars_re = b'[\w\d!#%&\'~_`><@,:/\$\*\+\-\.\^\|\)\(\?\}\{\=]'
 _cookie_re = re.compile(b"""(?x)
-    (?P<key>[^=]+)
-    \s*=\s*
+    (
+        (?P<key>[^=]+)
+        \s*=\s*
+    )? # when key is blank, browser leave out it and equals sign
```

Comment:
I think it would be better if the explanation would be "Without equals sign, the key is empty", plus a link to that SO answer.